### PR TITLE
Update Discv5 to the Latest Version

### PR DESCRIPTION
**Pull request by nisdas**
Part of #3147
- [x] Use the updated discv5 package.
- [x] Fix all references and breaking tests
- [x] Add tool to calculate ENR of a node

### CODE REVIEW

File: beacon-chain/p2p/discovery.go [hunk 1]
```diff
@@ -70,21 +92,18 @@ func convertToMultiAddr(nodes []*discv5.Node) []ma.Multiaddr {
...
-		return nil, errors.Wrap(err, "could not get pubkey from node ID")
-	}
+	pubkey := node.Pubkey()
 	assertedKey := convertToInterfacePubkey(pubkey)
 	id, err := peer.IDFromPublicKey(assertedKey)
 	if err != nil {
 		return nil, errors.Wrap(err, "could not get peer id")
 	}
-	multiAddrString := fmt.Sprintf("/ip4/%s/tcp/%d/p2p/%s", ip4.String(), node.TCP, id)
+	multiAddrString := fmt.Sprintf("/ip4/%s/tcp/%d/p2p/%s", ip4.String(), node.UDP(), id)
```
Comment (prestonvanloon) [inline beacon-chain/p2p/discovery.go]: UDP? This multiaddr is a TCP string

File: beacon-chain/p2p/discovery.go [hunk 2]
```diff
@@ -16,48 +17,69 @@ import (
...
-	listener := createListener(addr, int(cfg.Port), privKey)
-	bootNode, err := discv5.ParseNode(cfg.BootstrapNodeAddr)
+func createLocalNode(privKey *ecdsa.PrivateKey, ipAddr net.IP, port int) (*enode.LocalNode, error) {
+	db, err := enode.OpenDB("")
 	if err != nil {
-		return nil, err
-	}
-	if err := listener.SetFallbackNodes([]*discv5.Node{bootNode}); err != nil {
-		return nil, err
+		return nil, errors.Wrap(err, "Could not open node's peer database")
```
Comment (prestonvanloon) [inline beacon-chain/p2p/discovery.go]:
```suggestion
		return nil, errors.Wrap(err, "could not open node's peer database")
```

File: beacon-chain/p2p/discovery_test.go [hunk 3]
```diff
@@ -80,13 +75,13 @@ func TestStartDiscV5_DiscoverAllPeers(t *testing.T) {
 	}
 
 	// Wait for the nodes to have their local routing tables to be populated with the other nodes
-	time.Sleep(100 * time.Millisecond)
+	time.Sleep(1 * time.Second)
```
Comment (rauljordan) [inline beacon-chain/p2p/discovery_test.go]: Can this 1 * time.Second be put into a const at the top of the file? Easier to find and change in the future if needed

File: beacon-chain/p2p/discovery_test.go [hunk 4]
```diff
@@ -98,24 +93,22 @@ func TestStartDiscV5_DiscoverAllPeers(t *testing.T) {
...
-	_ = convertToMultiAddr([]*discv5.Node{node})
-
+	_ = convertToMultiAddr([]*enode.Node{node.Node()})
 	testutil.AssertLogsContain(t, hook, "node doesn't have an ip4 address")
 }
 
 func TestMultiAddrConversion_OK(t *testing.T) {
 	hook := logTest.NewGlobal()
-	port := 1024
+	_ = 1024
```
Comment (prestonvanloon) [inline beacon-chain/p2p/discovery_test.go]: ?

File: beacon-chain/p2p/discovery.go (hunk 1 above)
Comment (nisdas) [reply beacon-chain/p2p/discovery.go]: currently both udp and tcp connections are under the same port. For discv5 only their UDP port is specified not TCP. So if I use TCP I would get zero. I plan to address this in another PR, where we specify our separate UDP and TCP ports and then appropriately can add them to our multiaddr

File: tools/bootnode/bootnode_test.go [hunk 5]
```diff
@@ -17,44 +19,43 @@ func TestBootnode_OK(t *testing.T) {
...
 	}
+	listener := createListener(ipAddr, 4000, cfg)
+	defer listener.Close()
 
-	err = listener2.SetFallbackNodes([]*discv5.Node{listener.Self()})
+	cfg.PrivateKey = extractPrivateKey()
+	bootNode, err := enode.Parse(enode.ValidSchemes, listener.Self().String())
 	if err != nil {
-		t.Fatal(err)
+		log.Fatal(err)
```
Comment (prestonvanloon) [inline tools/bootnode/bootnode_test.go]: Revert this

**Comment by prestonvanloon** [comment]
This is blocked in cluster deployment pending some tooling for calculating the cluster ENR of the bootnode.

File: beacon-chain/p2p/discovery.go (hunk 1 above)
Comment (prestonvanloon) [reply beacon-chain/p2p/discovery.go]: OK please add a note / comment because this is a bit odd and another engineer may not have that context

**Pull request merged**